app/Http/Controllers/Api/CourseController.php

```diff
@@ -9,52 +9,114 @@
 use App\Models\CourseInstructor;
 use App\Models\CourseResources;
 use App\Models\CourseStudents;
+use App\Models\School;
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\File;
+use Auth;
 
 class CourseController extends Controller
 {
     public function create(Request $request)
     {
-        $course = new Course();
-
-        $validator = Validator::make($request->all(), [
-            'school_id' => 'required',
-            'faculty_id' => 'required',
+        //return dd($request->all());
+        $validator = Validator::make($request->all(), [                        
             'title' => 'required',
             'code' => 'required',
+            'resource_url' => 'required',
+            'experiment_id' => 'required',
+            'instructor_id' => 'required',
         ]);
-
+        $school_id = School::first();        
         if ($validator->fails()) {
             return response()->json(['error' => "All fields are required"], 400);
         }
 
-        $id = Util::uuid();
-        $school_id = $request->get('school_id');
-        $faculty_id = $request->get('faculty_id');
+
+        $course_uuid = Util::uuid();        
+        $faculty_id = 1;//please get the faculty id of the logged in user
         $title = $request->get('title');
         $code = $request->get('code');
         $description = $request->get('description');
         $status = $request->get('status') ?? 'Active';
 
-        $course->id = $id;
-        $course->school_id = $school_id;
-        $course->faculty_id = $faculty_id;
-        $course->title = $title;
-        $course->code = $code;
-        $course->description = $description;
-        $course->status = $status;
+        
+        $checkCourse = Course::where(['code' => $code])->first()->id;
 
-        $checkCourse = Course::where(['code' => $code])->first();
         if (empty($checkCourse)) {
-            if ($course->save()) {
-                return response()->json(['success' => true], 200);
+ 
+            
+            $course = array(
+            'id'          => $course_uuid,
+            'school_id'   => $school_id,
+            'faculty_id'  => $faculty_id,
+            'title'       => $title,
+            'code'        => $code,
+            'description' => $description,
+            'status'      => $status,
+            );
+            /*creating course resources*/
+            
+            $resource_uuid = Util::uuid();
+            $resourceUrl = $request->get('resource_url');
+
+            /*upload image*/
+            list($type, $img_data)  = explode(';', $resourceUrl);
+            list(, $img_data)  = explode(',', $img_data);
+            $decode_img = base64_decode($img_data);
+            $path = public_path().'/images/resources/';
+            if (!is_dir($path)){
+                File::makeDirectory($path,$mode =0777, true, true);
```

Comment:
We are not saving file on the system itself, rather using external source, right? @Auwalms

Reply:
Correct @bynalab. @saidabdul80 please revert this part of the PR. At the frontend level, you are expected to upload the resource (doc, pdf) to Amazon S3 (S3 Glacier gives us 10GB of free storage per month). Trying to not clutter our server or database with too many resources so the system doesn't get slow.

Reply:
@saidabdul80 please update the PR as discussed above.  just the changes to the resource URL BTW

Reply:
Ibid